### ISSUE TITLE
Subscribe TOPIC and Message Problem(solved)

### ISSUE THREAD

**Issue by housekorea** [aws/aws-iot-device-sdk-python#106]
Hello, I have face the problem which subscribe message

1) Raspberry PI python process publish every 10 second. 
2) It monitored well on the AWS IOT test subscriber panel(Same topic name)
3) At the same time, my macbook Subscribe process not capture fully. It looks seem like a dealy 30-40 seconds. It means that I published message 100(Consume 500 second), subscriber process took 12-13 message.

```
import os
import sys
from datetime import datetime
from AWSIoTPythonSDK.MQTTLib import AWSIoTMQTTClient
import time


CA_PATH = "/Users/house/Desktop/cert_pfc_0001/VeriSign-Class 3-Public-Primary-Certification-Authority-G5.pem"
PRIVATE_KEY_PATH ="/Users/house/Desktop/cert_pfc_0001/983f2f5c23142-private.pem.key"
CERTIFICATE_PATH ="/Users/house/Desktop/cert_pfc_0001/983f2f5c23142-certificate.pem.crt"

def test_cb(self, params, packet):
	print(packet.payload)
	f = open('/Users/house/Desktop/cert_pfc_0001/log.log','a+')
	f.write(packet.payload + '  /'+str(datetime.now()))
	f.write('\n')
	f.close()

myMQTTClient = AWSIoTMQTTClient("myClientID")

myMQTTClient.configureEndpoint("a3dijafieid469.iot.ap-northeast-2.amazonaws.com",8883)
myMQTTClient.configureCredentials(CA_PATH,PRIVATE_KEY_PATH,CERTIFICATE_PATH)




# AWSIoTMQTTClient connection configuration
myMQTTClient.configureAutoReconnectBackoffTime(1, 32, 20)
myMQTTClient.configureOfflinePublishQueueing(-1)  # Infinite offline Publish queueing
myMQTTClient.configureDrainingFrequency(2)  # Draining: 2 Hz
myMQTTClient.configureConnectDisconnectTimeout(10)  # 10 sec
myMQTTClient.configureMQTTOperationTimeout(5)  # 5 sec

print("Connect")
myMQTTClient.connect()
myMQTTClient.subscribe("pfc/test", 1, test_cb)

while True:
	None
```

**Comment by liuszeng** [comment]
Hi @housekorea ,

Thank you very much for using AWS IoT Device SDK for Python.

Can you provide more information regarding the issue you have? Is the SDK slow in receiving the messages or there is a message loss? Is there any log output that you can share when the issue happens? In the meantime, I will try to duplicate your issue on my side.

Thanks,
Liusu

**Comment by housekorea** [comment]
@liuszeng, Thank you : ) 

It is my fault. More detail about my environment, I did register one device on "AWS IoT" and download the certificate and public/private key files. And I copied this to other computer.It means that i shared one device ceritificate and keys on the two PC(Subscriber role / Publish role)!

On today morning, I find the AWS IoT policy following lines!

> When responding to a connection request, the message broker sends a CONNACK message. This message contains a flag to indicate if the connection is resuming a previous session. The value of this flag might be incorrect if two MQTT clients connect with the same client ID simultaneously.


((https://docs.aws.amazon.com/iot/latest/developerguide/protocols.html))

I catch this point, and try register two device and using each certificate, problem solved cleary!

Anyway, I have to read detail of the GUIDE on AWS!

Thank you.
